lib/rubocop/cop/style/redundant_line_continuation.rb

```diff
@@ -87,16 +88,20 @@ def on_new_investigation
 
         def require_line_continuation?(range)
           !ends_with_backslash_without_comment?(range.source_line) ||
-            string_concatenation?(range.source_line) ||
+            string_concatenation?(range) ||
             start_with_arithmetic_operator?(processed_source[range.line])
         end
 
         def ends_with_backslash_without_comment?(source_line)
           source_line.gsub(/#.+/, '').end_with?('\\')
         end
 
-        def string_concatenation?(source_line)
-          /["']\s*\\\z/.match?(source_line)
+        def string_concatenation?(range)
+          return true if /.*["']\s*\\\z/.match?(range.source_line)
+
+          processed_source.each_token.any? do |token|
+            ALLOWED_STRING_TOKENS.include?(token.type) && token.pos.overlaps?(range)
+          end
```

Comment:
This loop looks _a bit_ slow (but I have not measured), and it doesn’t really belong inside `#string_concatenation?`. Should it perhaps be a separate method (e.g. `#inside_string_literal?`?) and called _last_ inside `#require_line_continuation?`?
```rb
          !ends_with_backslash_without_comment?(range.source_line) ||
            string_concatenation?(range.source_line) ||
            start_with_arithmetic_operator?(processed_source[range.line]) ||
            inside_string_literal?(range)
```

Reply:
This looks like good logic ❤️ Thank you for the suggestion!